src/raster/rasterband.rs

```diff
@@ -884,6 +884,32 @@ impl<'a> RasterBand<'a> {
         }
     }
 
+    /// Set default raster histogram.
+    ///
+    /// # Arguments
+    ///
+    /// * `min` - Histogram lower bound
+    /// * `max` - Histogram upper bound
+    /// * `n_buckets` - Number of buckets in the histogram
+    /// * `counts` - Histogram values for each bucket
+    pub fn set_default_histogram(&self, min: f64, max: f64, counts: &mut [u64]) -> Result<()> {
+        let n_buckets = counts.len();
```

Comment:
Let's add an `assert!(n_buckets <= i32::MAX)` here and in the getter if missing.

(that matches `Dataset::rasterband`, there was a discussion on `Err` vs. `assert!` there)